lib/Entry.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE RankNTypes #-}
+{-# LANGUAGE TemplateHaskell #-}
+
+module Entry
+  ( dateModified
+  , Entry, EntryConvertible (..), newEntry
+  , path, masterField, fields
+  , Field (..), FieldKey (..)
+  , newField, newFieldKey, getFieldKey, newEntryTag, getEntryTag
+  , private, value, tags
+  )
+where
+
+
+import qualified Data.Text as T
+import qualified Data.HashMap.Strict as HS
+
+import Control.Lens
+import qualified Data.Aeson.Types as A
+import Data.Hashable (Hashable)
+import GHC.Generics (Generic)
+import Data.Time (UTCTime)
+import Language.Haskell.TH.Lens (_CaseE)
+
+type DateTime = UTCTime
+
+newtype FieldKey = FieldKey T.Text
+  deriving (Generic, Show, Eq)
+
+instance Hashable FieldKey
+
+instance A.ToJSON FieldKey where
+instance A.ToJSONKey FieldKey where
+instance A.FromJSON FieldKey where
+instance A.FromJSONKey FieldKey where
+
+newFieldKey :: T.Text -> Maybe FieldKey
+newFieldKey t =
+  if T.foldr ((&&) . (`elem` allowedChars)) True t then
+    Just $ FieldKey t
+  else
+    Nothing
+  where allowedChars = ['a'..'z'] ++ ['A'..'Z'] ++ ['0'..'9'] ++ "-_;:"
```

Comment:
As discussed [here](https://serokell.slack.com/archives/C024MA7K48Z/p1642788097019600?thread_ts=1642295588.004200&cid=C024MA7K48Z), can you please disallow the character `:` from appearing in field names?